### ISSUE TITLE
Improve assertion logging/messaging when operation does not return metadata

### ISSUE THREAD

**Issue by b-deam** [elastic/rally#1398]
If a user specifies an [`assertion`](https://esrally.readthedocs.io/en/stable/track.html?highlight=assertions#operations) inside an operation that does _not_ return any metadata (i.e. [the `raw-request` op](https://esrally.readthedocs.io/en/stable/track.html?highlight=assertions#raw-request)) the operation is marked as completing successfully in the console output. It's not made clear that the assertion was actually skipped - users instead have to go digging into the log file to check:

```
2021-12-07 05:53:32,367 ActorAddr-(T|:63625)/PID:86507 esrally.driver.runner DEBUG Skipping assertion check in [None] as [user-defined context-manager enabled runner for [raw-request]] does not return a dict.
```

Example track:
```
 $ cat track.json 
{
  "schedule": [
    {
        "operation": {
            "operation-type": "raw-request",
            "method": "GET",
            "path": "/.monitoring-es*/_search",
            "body": {
              "query": {
                "match_all": {}
              }
            },
          "detailed-results": true,
          "assertions": [
            {
              "property": "hits.total.value",
              "condition": "==",
              "value": "this-task-will-complete-successfully-because-theres-no-metadata-to-check"
            }
          ]
        }
    }
  ]
}%               
```

We should either improve the visibility of this by either making it clearer to the user via console logging, or throwing an exception (or a combination of the two).

**Comment by pquentin** [comment]
Agreed, only logging is a surprising decision here. Other parts raise errors, such as when `detailed-results` isn't true.

**Comment by riken127** [comment]
Hi, I may be missing some context here, but I noticed #1588 may already cover this issue.
From what I can see, Rally now fails with a `DataError` when it cannot evaluate an assertion because the runner did not return a `dict`.
There also seems to be a test covering this case called `test_raise_asserts_for_non_dicts`
Would it make sense to close this issue as fixed, or is there still a separate case here that I’m overlooking?

**Comment by pquentin** [comment]
Agreed, thank you for checking. This was fixed in https://github.com/elastic/rally/pull/1588, sorry for not closing this issue.
